pkg/apis/logging/v1/input_types.go

```diff
@@ -0,0 +1,44 @@
+package v1
+
+import (
+	"github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1/inputs"
+	sets "k8s.io/apimachinery/pkg/util/sets"
+)
+
+// Reserved input names.
+const (
+	InputNameApplication    = "application"    // Non-infrastructure container logs.
+	InputNameInfrastructure = "infrastructure" // Infrastructure containers and system logs.
+	InputNameAudit          = "audit"          // System audit logs.
+)
+
+var ReservedInputNames = sets.NewString(InputNameApplication, InputNameInfrastructure, InputNameAudit)
+
+// InputSpec defines a source of log messages.
+type InputSpec struct {
+	// Name used to refer to the input of a `pipeline`.
+	//
+	// +required
+	Name string `json:"name"`
```

Comment:
in the past we've had an alias Type for fields that expected specific values [1]. IIRC this follows an existing k8s pattern.

[1] https://github.com/openshift/cluster-logging-operator/blob/master/pkg/apis/logging/v1/clusterlogging_types.go#L252-L270

Reply:
I use that convention for all strings that are closed Enums, but these names are open-ended.
I dithered over using for input/output names since it provides some type-safety, where you try to use an InputName where an OutputName is expected. However it requires a bunch of duplication in code that deals with both generically.  It's probably worth doing, maybe raise a PR to rectify this later.

Reply:
Raise a PR to address that later, I think I should have done that for type-safety. I took the lazy way out as there's code that deals with both - but it would be easy enough to address.

Reply:
nit, but why not fix it while this pr is open? we can deal with duplication of code for now and clean that up later...